### PR TITLE
MIgrated Chuck MQTT to compose 

### PR DESCRIPTION
**Pull request by anubhav7nov**

### CODE REVIEW

**Comment by anubhav7nov** [comment]
@lokeshkalal @rizkyfadillah

File: chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                        color = Color(0xFF494A4A),
+                        style = TextStyle(
+                            fontWeight = FontWeight.Normal,
+                            fontSize = 14.sp
+                        )
+                    )
+
+                    Text(
+                        transmissionTime,
+                        color = Color(0xFF494A4A),
```
Comment (lokeshkalal) [inline chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt:161]: Lets use colors through theme to support dark mode

File: chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                        text = packetName,
+                        style = TextStyle(
+                            fontWeight = FontWeight.Bold,
+                            fontSize = 16.sp
+                        )
+                    )
+
+                    Text(
+                        text = packetInfo,
+                        color = Color(0xFF494A4A),
```
Comment (lokeshkalal) [inline chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt:152]: Lets use colors through theme to support dark mode

File: chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+@Composable
+internal fun TransactionList(
+    transactionList: List<MqttTransactionUiModel>,
+    intentLambda: (TransactionListIntent) -> Unit
+) {
+    LazyColumn {
+        items(transactionList) { item ->
+            TransactionListItem(item = item, intentLambda = intentLambda)
+            Divider(color = Color.LightGray)
```
Comment (lokeshkalal) [inline chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt:102]: Let's use colors through the theme to support dark mode, Similarly, all the other places where colors and icons which are dependent on theme

File: chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactiondetail/ui/fragment/TransactionDetailFragment.kt [hunk 4]
```diff
@@ -6,38 +6,34 @@ import android.content.Context
...
 import io.reactivex.rxkotlin.plusAssign
 import kotlin.reflect.KClass
-import kotlinx.android.synthetic.main.activity_transaction_detail.send_receive_view
-import kotlinx.android.synthetic.main.activity_transaction_detail.toolbar_title
 import kotlinx.android.synthetic.main.fragment_transaction_detail.copy
 import kotlinx.android.synthetic.main.fragment_transaction_detail.packet_body
 import kotlinx.android.synthetic.main.fragment_transaction_detail.packet_info
 import kotlinx.android.synthetic.main.fragment_transaction_detail.transaction_detail_loader
 
 internal class TransactionDetailFragment :
```
Comment (lokeshkalal) [inline chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactiondetail/ui/fragment/TransactionDetailFragment.kt:35]: Let's remove the fragment and use compose navigation?

File: chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt (hunk 1 above)
Comment (rizkyfadillah) [reply chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt:161]: Can we define the colors first for both Light and Dark mode? I can see the Color Palette that you have defined here, but:
- Both `primaryVariant` and `secondary` colors are same for Light and Dark (while they should be different based on Light or Dark mode)
- Not enough contrast between Light's `primary` with Dark's

Screen Shot 2022-06-14 at 13 05 36

Reference: https://material.io/design/color/the-color-system.html#color-theme-creation
 
Once we have the colors, then we can make use of it on Composables.

**Comment by rizkyfadillah** [comment]
@anubhav7nov @lokeshkalal To address the comments in this PR, should I push the changes to the merging branch or should I create a new branch and submit a new MR against `chuck-mqtt-compose`?

**Comment by rizkyfadillah** [comment]
> @anubhav7nov @lokeshkalal To address the comments in this PR, should I push the changes to the merging branch (`chuck-mqtt-compose`) or should I create a new branch and submit a new MR against `chuck-mqtt-compose`?

Raised another PR here https://github.com/gojek/courier-android/pull/30 (pointing to `chuck-mqtt-compose`) to address these:
1. Add missing search functionality using Compose (https://github.com/gojek/courier-android/pull/20#discussion_r894479707)
2. Introduce `navigation-compose` library and remove `TransactionDetailFragment` (https://github.com/gojek/courier-android/pull/20#discussion_r894450212)
3. Apply proper material colors to support Dark and Light mode in Compose (https://github.com/gojek/courier-android/pull/20#discussion_r894442689)

**Pull request closed without merging**